eclass/libretro-core.eclass

```diff
@@ -0,0 +1,168 @@
+
+# @FUNCTION: libretro-core_src_unpack
+# @DESCRIPTION:
+# The libretro-core src_unpack function which is exported.
+#
+# This function retrieves the remote Libretro core info files.
+libretro-core_src_unpack() {
+	# If this is a live ebuild, retrieve this core's remote repository.
+	if [[ ${PV} == *9999 ]]; then
+		git-r3_src_unpack
+		# Add used commit SHA for version information, the above could also work.
+		LIBRETRO_COMMIT_SHA=$(git -C "${EGIT3_STORE_DIR}/${LIBRETRO_REPO_NAME//\//_}.git" rev-parse HEAD)
+	# Else, unpack this core's local tarball.
+	else
+		default_src_unpack
+	fi
+}
+
+# @FUNCTION: libretro-core_src_prepare
+# @DESCRIPTION:
+# The libretro-core src_prepare function which is exported.
+#
+# This function prepares the source by making custom modifications.
+libretro-core_src_prepare() {
+	local flags_modified=0
+	ebegin "Attempting to hack Makefiles to use custom-cflags"
+	for makefile in "${S}"/?akefile* "${S}"/target-libretro/?akefile*; do
+		# * Convert CRLF to LF
+		# * Expand *FLAGS to prevent potential self-references
+		# * Where LDFLAGS directly define the link version 
+		#   script append LDFLAGS and LIBS
+		# * Where SHARED is used to provide shared linking
+		#   flags ensure final link command includes LDFLAGS
+		#   and LIBS
+		# * Always use $(CFLAGS) when calling $(CC)
+		sed \
+			-e 's/\r$//g' \
+			-e "/flags.*=/s/-O[[:digit:]]/${CFLAGS}/g" \
+			-e "/CFLAGS.*=/s/-O[[:digit:]]/${CFLAGS}/g" \
+			-e "/.*,--version-script=.*/s/$/ ${LDFLAGS} ${LIBS}/g" \
+			-e "/\$(CC)/s/\(\$(SHARED)\)/\1 ${LDFLAGS} ${LIBS}/" \
+			-e 's/\(\$(CC)\)/\1 \$(CFLAGS)/g' \
+			-i "${makefile}" \
+			&> /dev/null && flags_modified=1
+	done
+	[[ ${flags_modified} == 1 ]] && true || false
+	eend $?
+	export OPTFLAGS="${CFLAGS}"
+
+	# Populate COMMIT for GIT_VERSION
+	if [[ -z "${CUSTOM_LIBRETRO_COMMIT_SHA}" ]]; then
+		CUSTOM_LIBRETRO_COMMIT_SHA="\" ${LIBRETRO_COMMIT_SHA:0:7}\""
+	fi
+
+	for makefile in "${S}"/?akefile* "${S}"/target-libretro/?akefile*; do
+		# Add short-rev to Makefile
+		sed \
+			-e "s/GIT_VERSION\s.=.*$/GIT_VERSION=${CUSTOM_LIBRETRO_COMMIT_SHA}/g" \
+			-i "${makefile}" \
+			&> /dev/null
+	done
+	default_src_prepare
+}
+
+# @FUNCTION: libretro-core_src_compile
+# @DESCRIPTION:
+# The libretro-core src_compile function which is exported.
+#
+# This function compiles the shared library for this Libretro core.
+libretro-core_src_compile() {
+	emake CC=$(tc-getCC) CXX=$(tc-getCXX) \
+		$(usex debug "DEBUG=1" "") "${myemakeargs[@]}" \
+		$([[ -f makefile.libretro ]] && echo '-f makefile.libretro') \
+		$([[ -f Makefile.libretro ]] && echo '-f Makefile.libretro')
+}
+
+# @FUNCTION: libretro-core_src_install
+# @DESCRIPTION:
+# The libretro-core src_install function which is exported.
+#
+# This function installs the shared library for this Libretro core.
+libretro-core_src_install() {
+	# Absolute path of the directory containing Libretro shared libraries.
+	LIBRETRO_LIB_DIR="${EPREFIX}/usr/$(get_libdir)/libretro"
```

Comment:
This was actually correct without `EPREFIX`; `insinto` automatically adds EPREFIX to the path `EPREFIX`. Please revert the change.

Reply:
Done.

The documentation at https://devmanual.gentoo.org/function-reference/install-functions/index.html should probably be updated. I, at least, get tripped up frequently on the use of EPREFIX as related to the install functions.